test/test_import.py

```diff
@@ -0,0 +1,324 @@
+import unittest
+import os
+
+import psycopg2
+import psycopg2.extras
+import decimal
+from time import sleep
+
+from utils import DbTestBase
+
+class TestTriggers(unittest.TestCase, DbTestBase):
+
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.conn.rollback()
+
+
+    @classmethod
+    def setUpClass(cls):
+        pgservice=os.environ.get('PGSERVICE')
+        if not pgservice:
+          pgservice='pg_qgep'
+        cls.conn = psycopg2.connect("service={service}".format(service=pgservice))
+
+
+    # - correct update with 1 old outlet and 1 new outlet and 0 old inlet and 0 new inlet
+    #   -> updated structure
+    #   -> updated reach
+    #   -> updated reach_point
+    #   -> deleted in quarantene
+    def test_update_with_outlet(self):
+        # obj_id from the test data
+        obj_id = 'ch13p7mzMA000012'
+
+        # change remark from 'Strasseneinlauf' to 'Strassenauslauf'
+        # change co_material from 233 to 3015
+        row = {
+                'remark': 'Strassenauslauf',
+                'co_material': 3015,
+                'outlet_1_material': 5081,
+                'outlet_1_clear_hight': 160,
+                'outlet_1_depth_m': 100,
+                'photo1' : 'funky_selfie.png'
+        }
+
+        # update
+        self.update('vw_manhole', row, obj_id, 'qgep_import')
+
+        # it should be in the live table qgep_od.wastewater_structure
+        row = self.select( 'wastewater_structure', obj_id, 'qgep_od')
+        self.assertIsNotNone( row )
+        self.assertEqual( row['remark'], 'Strassenauslauf')
+
+        # it should be in the view qgep_import.vw_manhole
+        row = self.select( 'vw_manhole', obj_id, 'qgep_import')
+        self.assertIsNotNone( row )
+        self.assertEqual( row['co_material'], 3015)
+        self.assertEqual( row['remark'], 'Strassenauslauf')
+
+        # it should be in the live table qgep_od.reach and qgep_od.reach_point
+        cur = self.cursor()
+        cur.execute("SELECT re.material, re.clear_height, rp.level, ws.co_level\
+            FROM {schema}.reach re\
+            LEFT JOIN {schema}.reach_point rp ON rp.obj_id = re.fk_reach_point_from\
+            LEFT JOIN {schema}.wastewater_networkelement wn ON wn.obj_id = rp.fk_wastewater_networkelement\
+            LEFT JOIN {schema}.vw_qgep_wastewater_structure ws ON ws.obj_id = wn.fk_wastewater_structure\
+            WHERE ws.obj_id = '{obj_id}'".format(schema='qgep_od', obj_id=obj_id))
```

Comment:
rather than `execute('...{var1}....{var2}...'.format(var1=...,var2=...))` you should prefer using integrated variables handling:
http://initd.org/psycopg/docs/cursor.html#execute
`execute('....%(var1)s....%(var2)I....', var1=..., var2=...)`

Reply:
not a big deal for tests, but good to know

Reply:
Haven't knew that. Thanks.

Reply:
This does only work for variable names because it creates single quotes around it.
See http://initd.org/psycopg/docs/usage.html#query-parameters
>> Only query values should be bound via this method: it shouldn’t be used to merge table or field names to the query (Psycopg will try quoting the table name as a string value, generating invalid SQL).
So I could do it with `psycopg2.sql` http://initd.org/psycopg/docs/sql.html#module-psycopg2.sql. It's working from psycopg2 ver. 2.7 on. Do you see a problem with that?

Reply:
`obj_id` is a variable, no?

Reply:
Yes. It's only about the `obj_id`?
So: 
```
        cur.execute("SELECT *\
            FROM {schema}.file\
            WHERE object = '%(obj_id)s'".format(schema='qgep_od'), {'obj_id':obj_id})
```
Or with psycopg2.sql like this:
```
        cur.execute(sql.SQL("SELECT *\
            FROM {schema}.file\
            WHERE object = '{obj_id}'").format(schema=sql.Identifier('qgep_od'), obj_id=sql.Identifier(obj_id)))
```
What is the preferable way?

Reply:
Sorry as the second I mean the combi of them:
```
        cur.execute(sql.SQL("SELECT *\
            FROM {schema}.file\
            WHERE object = %(obj_id)s").format(schema=sql.Identifier('qgep_od')), {'obj_id':obj_id}))
```

Reply:
I think so. The schema is a hardcoded value, which is mostly here for code readability and maintainability. I'd go for option 1, I assume it will have a higher probability for things like prepared statements because it transfers the knowledge of which parts are variable to psycopg2.

Reply:
Update sounds good, should also prevent from SQL injection if the schema is ever configurable.

Reply:
Done